### PR TITLE
Define a few undefined behaviors related to DrawBuffers.

### PR DESCRIPTION
**Pull request by zhenyao**
Also add a few test cases for that.

### CODE REVIEW

**Comment by zhenyao** [comment]
@Kangz @kenrussell @jdashg @grorg @RafaelCintron PTAL

File: sdk/tests/conformance/extensions/webgl-draw-buffers.html [hunk 1]
```diff
@@ -300,6 +305,14 @@
   wtu.checkCanvas(gl, [0, 255, 0, 255], "should be green");
   gl.deleteProgram(program);
   wtu.glErrorShouldBe(gl, gl.NO_ERROR, "there should be no errors");
+
+  debug("");
+  debug("test that gl_MaxDrawBuffers is frozen at link time and enabling extenion won't change it.");
```
Comment (kenrussell) [inline sdk/tests/conformance/extensions/webgl-draw-buffers.html]: typo: extenion -> the extension

File: specs/latest/2.0/index.html [hunk 2]
```diff
@@ -2809,6 +2809,20 @@ <h2 class="no-toc">Table of contents</h2>
...
+    <p>
+        If a fragment shader writes to none of <code>gl_FragColor</code>, <code>gl_FragData</code>, nor any
+        user-defined output variables, the values of the fragment colors following shader execution are untouched.
+        If some, but not all user-defined output variables are written, the values of fragment colors corresponding
+        to unwritten variables are similarly untouched.
+    </p>
+
+    <p>
+        If the values written by the fragment shader do not match the format(s) of the corresponding
+        color buffer(s), and the color buffer states are not set to <code>NONE</code> by <code>DrawBuffers</code>,
```
Comment (kenrussell) [inline specs/latest/2.0/index.html]: Is it worth adding some non-normative text with an example of this mismatch? For example, clearly stating that the shader attempts to write an integer value to a floating-point color buffer, or vice versa?

**Comment by kenrussell** [comment]
LGTM from my standpoint. A couple of comments.

File: specs/latest/2.0/index.html (hunk 2 above)
Comment (zhenyao) [inline specs/latest/2.0/index.html]: Good suggestion. Done.

File: sdk/tests/conformance/extensions/webgl-draw-buffers.html (hunk 1 above)
Comment (zhenyao) [inline sdk/tests/conformance/extensions/webgl-draw-buffers.html]: Done

**Comment by zhenyao** [comment]
Addressed kbr's comments.  Merging now.

Others, specifically @jdashg , if you feel differently, we can revise later.

**Pull request merged**